packages/widget/src/context/wallet.ts

```diff
@@ -31,37 +33,70 @@ export enum WalletContextKeys {
   SUBSTRATE_WALLET = 'substrateWallet'
 }
 
+export interface SubstrateProviderContext {
+  substrateProviders?: Array<{ domainId: number; api: ApiPromise }>;
+  selectedProvider?: ApiPromise;
+}
+
 declare global {
   interface HTMLElementEventMap {
     walletUpdate: WalletUpdateEvent;
+    substrateProviderUpdate: SubstrateProviderUpdateEvent;
   }
 }
 
 export const walletContext = createContext<WalletContext>(
   Symbol('sygma-wallet-context')
 );
 
+export const substrateProviderContext = createContext<SubstrateProviderContext>(
+  Symbol('substrate-provider-context')
+);
+
 export class WalletUpdateEvent extends CustomEvent<WalletContext> {
   constructor(update: Partial<WalletContext>) {
     super('walletUpdate', { detail: update, composed: true, bubbles: true });
   }
 }
 
+export class SubstrateProviderUpdateEvent extends CustomEvent<SubstrateProviderContext> {
+  constructor(update: Partial<SubstrateProviderContext>) {
+    super('substrateProviderUpdate', {
+      detail: update,
+      composed: true,
+      bubbles: true
+    });
+  }
+}
+
 @customElement('sygma-wallet-context-provider')
 export class WalletContextProvider extends BaseComponent {
   //TODO: add properties to allow widget to pass external provider/signers.
 
   @provide({ context: walletContext })
   private walletContext: WalletContext = {};
 
+  @provide({ context: substrateProviderContext })
+  private substrateProviderContext: SubstrateProviderContext = {};
+
   @property({ attribute: false, type: Object })
   evmWalllet?: EvmWallet;
 
+  @property({ attribute: false })
+  substrateProviders?: Array<{ domainId: number; api: ApiPromise }> = [];
```

Comment:
I don't think user should care about our internal domain id. He should just provide list of ApiPromise and we should from each fetch chainId and create map